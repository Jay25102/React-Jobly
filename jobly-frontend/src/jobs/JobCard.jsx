import React, { useState, useContext, useEffect } from "react";
import UserContext from "../forms/UserContext";
import "./JobCard.css";

/**
 * Display info about each job. Rendered from JobCardList.
 */
function JobCard({id, title, salary, equity, companyName}) {
    const { hasAppliedToJob, applyToJob } = useContext(UserContext);
    const [applied, setApplied] = useState();
    console.debug("JobCard");

    useEffect(function updateAppliedStatus() {
        console.debug("JobCard useEffect updateAppliedStatus", "id=", id);
        setApplied(hasAppliedToJob(id));
    }, [id, hasAppliedToJob]);

    // apply for a job if not already applied
    async function handleApply(e) {
        if (hasAppliedToJob(id)) return;
        applyToJob(id);
        setApplied(true);
    }

    // make displaying salary look nicer in text
    function formatSalary(salary) {
        const digitsRev = [];
        const salaryStr = salary.toString();

        for (let i = salaryStr.length - 1; i >= 0; i--) {
            digitsRev.push(salaryStr[i]);
            if(i > 0 && i % 3 === 0) digitsRev.push(",");
        }

        return digitsRev.reverse().join("");
    }

    return (
        <div className="JobCard card">
            <div className="card-body">
                <div className="card-title">Title: {title}</div>
                <p>{companyName}</p>
                {salary && <div><small>Salary: {formatSalary(salary)}</small></div>}
                {equity !== undefined && <div><small>Equity: {equity}</small></div>}
                <button
                    className="btn btn-danger font-weight-bold text-uppercase float-right"
                    onClick={handleApply}
                    disabled={applied}
                >{applied ? "Applied" : "Apply"}</button>
            </div>
        </div>
    )
}

export default JobCard;