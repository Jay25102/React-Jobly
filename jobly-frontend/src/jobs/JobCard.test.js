import React from "react";
import { render } from "@testing-library/react";
import JobCard from "./JobCard";
import { UserProvider } from "../testUtils";

describe("JobCard", function() {
    test("it renders without crashing", function() {
        render(
            <UserProvider>
                <JobCard/>                
            </UserProvider>
    );
    });
    test("it matches snapshot", function() {
        let item = { title: "title", salary: 55555, equity: 55};
        const { asFragment } = render(
            <UserProvider>
                <JobCard item={item}/>
            </UserProvider>
        );
        expect(asFragment()).toMatchSnapshot();
    });
});
