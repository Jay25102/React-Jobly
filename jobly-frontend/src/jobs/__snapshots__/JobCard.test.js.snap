// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JobCard it matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <div>
      Applied: 
    </div>
    <div>
      Title: 
    </div>
    <div>
      Company: 
    </div>
    <button>
      Apply
    </button>
  </div>
</DocumentFragment>
`;
